src/main/kotlin/org/jetbrains/research/ml/ast/gumtree/diff/PsiElementTransformer.kt

```diff
@@ -11,45 +11,48 @@ import org.jetbrains.research.ml.ast.gumtree.tree.Numbering
 import org.jetbrains.research.ml.ast.gumtree.tree.Numbering.PsiTreeUtils.Companion.id
 import org.jetbrains.research.ml.ast.gumtree.tree.Numbering.PsiTreeUtils.Companion.setId
 
-class PsiElementTransformer(
-    project: Project,
-    srcPsi: PsiElement,
-    dstPsi: PsiElement,
-    private val numbering: Numbering
+data class PsiTransformation(
+    private val srcPsi: PsiElement,
+    private val dstPsi: PsiElement,
+    private val numbering: Numbering,
+    var srcPsiNodes: MutableList<PsiElement> = ArrayList(),
+    var dstPsiNodes: MutableList<PsiElement> = ArrayList(),
+    val insertedNodesIds: MutableSet<Int> = HashSet()
```

Comment:
Why do we need to pass srcPsiNodes, dstPsiNodes, and insertedNodesIds to the constructor? what if we pass the wrong variables? why not make fields and count automatically?